staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -131,21 +131,27 @@ func (rt *Transport) WrappedRoundTripper() http.RoundTripper {
 
 // rewriteURL rewrites a single URL to go through the proxy, if the URL refers
 // to the same host as sourceURL, which is the page on which the target URL
-// occurred. If any error occurs (e.g. parsing), it returns targetURL.
-func (t *Transport) rewriteURL(targetURL string, sourceURL *url.URL) string {
+// occurred, or if the URL matches the sourceHost. If any error occurs (e.g.
+// parsing), it returns targetURL.
+func (t *Transport) rewriteURL(targetURL string, sourceURL *url.URL, sourceHost string) string {
```

Comment:
nit: It was a little confusing for me to read `sourceURL.host` vs `sourceHost`. It is much clear to call it `sourceReqHost` or `sourceRequestHost`

Reply:
Agree that the names are almost impenetrable

Reply:
Updated.